packages/button/src/ButtonContent/ButtonContent.tsx

```diff
@@ -1,30 +1,38 @@
 import React, { useEffect, useRef } from 'react';
 
-import { css, cx } from '@leafygreen-ui/emotion';
+import { cx } from '@leafygreen-ui/emotion';
 import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { Spinner } from '@leafygreen-ui/loading-indicator';
 import { registerRipple } from '@leafygreen-ui/ripple';
 
-import ButtonIcon from '../ButtonIcon/ButtonIcon';
 import { ButtonProps, Size, Variant } from '../types';
 
 import {
   buttonContentSizeStyle,
   buttonContentStyle,
+  buttonSpinnerSize,
+  centeredSpinnerContainerStyles,
+  centeredSpinnerStyles,
+  hiddenContentStyles,
   rippleColors,
   rippleStyle,
+  spinnerColor,
 } from './ButtonContent.styles';
+import DefaultContent from './DefaultContent';
 
 type ButtonContentProps = Omit<ButtonProps, 'as'>;
 
-export const ButtonContent = ({
-  rightGlyph,
-  leftGlyph,
-  darkMode: darkModeProp,
-  disabled = false,
-  variant = Variant.Default,
-  size = Size.Default,
-  children,
-}: ButtonContentProps) => {
+export const ButtonContent = (props: ButtonContentProps) => {
+  const {
+    darkMode: darkModeProp,
```

Comment:
nit: Just a thought, but i think if inside `Button` we wrap `ButtonContent` with `LeafygreenProvider` we can avoid passing down the darkMode prop.